script/update-docs.rb

```diff
@@ -465,14 +466,15 @@ def index_doc(filter_tags, doc_list, get_content)
         end
         # HTML anchor on hdlist1 (i.e. command options)
         html.gsub!(/<dt class="hdlist1">(.*?)<\/dt>/) do |_m|
-          text = $1.tr("^A-Za-z0-9-", "")
+          inner_html = $1
+          text = inner_html.gsub(/<[^>]+>/, "").tr("^A-Za-z0-9-", "")
           # use txt_path for backward compatibility of already-existing
           # deep links shared across the interwebs.
           anchor = "#{txt_path}-#{text}"
           # handle anchor collisions by appending -1
           anchor += "-1" while ids.include?(anchor)
           ids.add(anchor)
-          "<dt class=\"hdlist1\" id=\"#{anchor}\"> <a class=\"anchor\" href=\"##{anchor}\"></a>#{$1} </dt>"
+          "<dt class=\"hdlist1\" id=\"#{anchor}\"> <a class=\"anchor\" href=\"##{anchor}\"></a>#{inner_html} </dt>"
```

Comment:
Here, I was thinking more of something like:
```ruby
stripped_id = Nokogiri::HTML.parse(m).tr("^A-Za-z0-9-", "")
"<dt class=\"hdlist1\" id=\"#{anchor}\"> <a class=\"anchor\" id=\"#{stripped_id}\" href=\"##{stripped_id}\"></a>#{inner_html} </dt>"
```

This way, we retain the historical ids, but we know advertise a more understandable anchor name (without the "gt" and "lt" garbage). 

Note: it is just an idea, I haven't tested if it works for all edge cases.

Reply:
Good idea!